# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/u2f/tests/frame_override_request.html
# Commit: f7937d3264db
# Full Hash: f7937d3264db00771b46cb1fcba71640d8df05cb
# Author: J.C. Jones <jjones@mozilla.com>
# Date: 2019-03-30 09:32:04
# Regressor Bug: 1448408
# File Overlap Count: 2
# Description:
#   Bug 1448408 - Web Authentication - Don't immediately abort on visibility events r=keeler
#   
#   The published recommendation of L1 for WebAuthn changed the visibility/focus
#   listening behaviors to a SHOULD [1], and Chromium, for reasons like our SoftU2F
#   bug [0], opted to not interrupt on tabswitch/visibility change.
# ==============================================================================

diff -r a5c4f7e6069a -r f7937d3264db dom/u2f/tests/frame_override_request.html
--- a/dom/u2f/tests/frame_override_request.html	Fri Mar 29 19:38:53 2019 +0000
+++ b/dom/u2f/tests/frame_override_request.html	Fri Mar 29 17:59:08 2019 +0000
@@ -61,21 +61,23 @@
       // Request a new credential.
       await requestMakeCredential("aborted1");
 
-      // Request another credential, the first request will abort.
+      // Request another credential, the new request will abort.
       await requestMakeCredential("aborted2");
-      local_is(status, "aborted1", "first request aborted");
-
-      // Request an assertion, the second request will abort.
-      await requestGetAssertion("aborted3");
       local_is(status, "aborted2", "second request aborted");
 
-      // Request another assertion, the third request will abort.
-      await requestGetAssertion("aborted4");
+      // Request an assertion, the new request will still abort.
+      await requestGetAssertion("aborted3");
       local_is(status, "aborted3", "third request aborted");
 
-      // Request another credential, the fourth request will abort.
+      // Request another assertion, this fourth request will abort.
+      await requestGetAssertion("aborted4");
+      local_is(status, "aborted4", "fourth request aborted");
+
+      // Request another credential, the fifth request will still abort. Why
+      // do we keep trying? Well, the test originally looked like this, and
+      // let's face it, it's kinda funny.
       await requestMakeCredential("aborted5");
-      local_is(status, "aborted4", "fourth request aborted");
+      local_is(status, "aborted5", "fifth request aborted");
 
       local_finished();
     })();